docs/source/guide/guide-executors.rst

```diff
@@ -413,12 +413,93 @@ in this example can be found `here <https://quantumcomputing.stackexchange.com/a
 
 Qiskit: Density-matrix Simulation with Depolarizing Noise
 -----------------------------------------------------------
-    TODO
+
+Analogous to the Cirq implementation, this executor can be used for noisy depolarizing
+simulation.
+
+.. testcode::
+
+    import qiskit
+    from qiskit import QuantumCircuit
+    import numpy as np
+    import copy
+
+    # Noise simulation packages
+    from qiskit.providers.aer.noise import NoiseModel
+    from qiskit.providers.aer.noise.errors.standard_errors import depolarizing_error
+    from qiskit.providers.aer.extensions import snapshot_density_matrix
+
+    QISKIT_SIMULATOR = qiskit.Aer.get_backend("qasm_simulator")
+
+    def execute(circ: QuantumCircuit, obs: np.ndarray, noise: float) -> float:
+        """Simulates the evolution of the noisy circuit and returns
+        the expectation value of the observable.
+
+        Args:
+            circ: The input Cirq circuit.
+            obs: The observable to measure as a NumPy array.
+            noise: The depolarizing noise strength as a float, i.e. 0.001 is 0.1%.
+
+        Returns:
+            The expectation value of obs as a float.
+        """
+        if len(circ.clbits) > 0:
+            raise ValueError("This executor only works on programs with no classical bits.")
+
+        circ = copy.deepcopy(circ)
+        # we need to modify the circuit to measure obs in its eigenbasis
+        # we do this by appending a unitary operation
+        eigvals, U = np.linalg.eigh(obs) # obtains a U s.t. obs = U diag(eigvals) U^dag
+        circ.unitary(np.linalg.inv(U), qubits=range(circ.num_qubits))
+        circ.snapshot_density_matrix('final')
+        circ.measure_all()
```

Comment:
Hi @Aaron-Robertson, thank you very much for this contribution!
I wonder if we should remove 
```suggestion
```
I wonder if one should remove this block of code. This was used to map the observable to the computational basis of the qubits. However I think we don't have to apply U if we directly use the density matrix.

Reply:
Hey @andreamari, right you are! And you're welcome. I'm happy to be involved and appreciate the opportunity for feedback.

I can make the change on this executor as soon as I have a chance (though I'll leave the `circ.snapshot_density_matrix('final')`, which I snuck in there).

What about the sampled case? Better to invoke the density matrix snapshot directly I think, but should it be a separate issue?

Reply:
I suppose they are examples, and nothing wrong with showing more than one way to do a thing. I could just add a note for the distinction.

Reply:
The qiskit noisy executor with a finite number of shots is already in the docs and I think it is fine. In that case I would keep the change of basis (i.e. the unitary U) since that executor, even if simulated, should mimic the hardware execution where the density matrix is not available.